### ISSUE TITLE
Deprecate `deferredGenerator`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6044]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6044|
|Type|task|
|Created|2012-10-03 13:35:05Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-deferredGenerator-6044-4|

`twisted.internet.defer.deferredGenerator` is like `twisted.internet.defer.inlineCallbacks` but for Python <= 2.5.  Since Twisted no longer supports Python 2.5, there's no reason to continue to offer `deferredGenerator` (`inlineCallbacks` is better in every way except for the newer Python requirement).

<details><summary>Searchable metadata</summary>

```
trac-id__6044 6044
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_deprecate_deferredGenerator_6044_4 branches/deprecate-deferredGenerator-6044-4
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349271305000000 1349271305000000
changetime__1397539177000000 1397539177000000
version__None None
owner__tom_prince tom.prince

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince commented|
|-|-|

(In [37164]) Branching to deprecate-deferredGenerator-6044.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince set owner to @tomprince|
|-|-|

This depends on [#6316](https://github.com/twisted/twisted/issues/6316) and [#5564](https://github.com/twisted/twisted/issues/5564).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince commented|
|-|-|

(In [39166]) Branching to deprecate-deferredGenerator-6044-2.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince commented|
|-|-|

(In [39171]) Branching to deprecate-deferredGenerator-6044-3.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

I split off the removal of existing uses to [#6626](https://github.com/twisted/twisted/issues/6626). Once that is merged, this is ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince commented|
|-|-|

(In [40920]) Branching to deprecate-deferredGenerator-6044-4.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

(In [40922]) Branching to deprecate-deferredGenerator-6044-4.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph set owner to @tomprince|
|-|-|

Basically; cool.  Please [make twistedchecker happy](https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/1557/steps/run-twistedchecker/logs/new%20twistedchecker%20errors) though.  Also, any reason `@deprecated` didn't work for the `__init__` thing?  Is it the `stacklevel` thing?  Perhaps there should be a ticket.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_10"></a>|@tomprince commented|
|-|-|

[#7147](https://github.com/twisted/twisted/issues/7147) is a ticket about the deprecation of `waitForDeferred`.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince set status to `closed`|
|-|-|

(In [42242]) Merge deprecate-deferredGenerator-6044-4: Deprecate deferredGenerator.

Author: tom.prince
Reviewer: glyph
Fixes: [#6044](https://github.com/twisted/twisted/issues/6044)

twisted.internet.defer.deferredGenerator is like
twisted.internet.defer.inlineCallbacks but for Python <= 2.5. Since
Twisted no longer supports Python 2.5, there's no reason to continue
to offer deferredGenerator (inlineCallbacks is better in every way
except for the newer Python requirement).
